docs/tutorials/ml_inference/language_identification/ml_inference_with_language_identification_ingest.md

```diff
@@ -0,0 +1,468 @@
+            "Effect": "Allow",
+            "Resource": "arn:aws:sagemaker:your_region:your_account_id:endpoint/your_sagemaker_endpoint_name"
+        }
+    ]
+}
+```
+Generate an access key and secret key for this user.
+
+#### b. Create the Connector
+Use the IAM user's credentials to create the connector.
+
+```json
+POST /_plugins/_ml/connectors/_create
+{
+  "name": "Sagemaker language identification model connector",
+  "description": "Connector for language identification model on SageMaker",
+  "version": 1,
+  "protocol": "aws_sigv4",
+  "credential": {
+    "aws_access_key_id": "your_aws_access_key_id",
+    "aws_secret_access_key": "your_aws_secret_access_key"
+  },
+  "parameters": {
+    "region": "your_aws_region",
+    "service_name": "sagemaker"
+  },
+  "actions": [
+    {
+      "action_type": "predict",
+      "method": "POST",
+      "url": "https://runtime.sagemaker.your_aws_region.amazonaws.com/endpoints/your_sagemaker_endpoint_name/invocations",
+      "headers": {
+        "content-type": "application/json"
+      },
+      "request_body": "{"inputs":"${parameters.inputs}"}"
+    }
+  ]
+}
+```
+Sample response:
+```json
+{
+  "connector_id": "bLYgeJEBXV92Z6od5rXM"
+}
+```
+Copy the `connector_id` from the response.
+
+## 3. Register and Deploy the Model in OpenSearch
+
+Use the `connector_id` from the previous step to register the model in OpenSearch.
+
+```json
+POST /_plugins/_ml/models/_register
+{
+  "name": "sagemaker-language-identification",
+  "version": "1",
+  "function_name": "remote",
+  "description": "Remote model for language identification",
+  "connector_id": "your_connector_id"
+}
+```
+Sample response:
+```json
+{
+  "task_id": "hbYheJEBXV92Z6oda7Xb",
+  "status": "CREATED",
+  "model_id": "hrYheJEBXV92Z6oda7X7"
+}
+```
+Copy the `model_id` and deploy the model.
+
+```json
+POST /_plugins/_ml/models/your_model_id/_deploy
+```
+
+## 4. Test the Remote Model
+You can now test the model directly to ensure it's working correctly.
+
+```json
+POST /_plugins/_ml/models/your_model_id/_predict
+{
+  "parameters": {
+    "inputs": "Say this is a test"
+  }
+}
+```
+The response should show the predicted language. Note the path to the label, `response[0].label`, which we will use in the ingest pipeline.
+```json
+{
+  "inference_results": [
+    {
+      "output": [
+        {
+          "name": "response",
+          "dataAsMap": {
+            "response": [
+              {
+                "label": "en",
+                "score": 0.9411176443099976
+              }
+            ]
+          }
+        }
+      ]
+    }
+  ]
+}
+```
+
+---
+
+# Building an Ingest Pipeline for Language Identification
+
+## 1. Create an Ingest Pipeline
+
+This pipeline uses the `ml_inference` processor to identify the language of the `name` and `notes` fields. It then uses a `copy` processor to create a new, language-specific field (e.g., `name_en`, `name_de`).
+
+```json
+PUT _ingest/pipeline/language_identification_pipeline
```

Comment:
Do you think it's necessary to add a some general guidance of this ml_inference processor like 1) the number of the copy sub-processor should match the number of entries in the "input_map" and 2) the response is an array which matches to the response in dataAsMap, etc. So the CX can use this general guideline to avoid mistakes when configuring their own ml_inference processor?

Reply:
added explanation in the new commit [add explainations for input and output mapping](https://github.com/opensearch-project/ml-commons/pull/3966/commits/5b85e280d0c62c3a6868b71b2be9a24eacc77186)